app/Console/Commands/CreateUsers.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+  namespace App\Console\Commands;
+
+  use App\Models\User;
+  use Carbon\Carbon;
+  use Exception;
+  use Illuminate\Console\Command;
+  use Illuminate\Support\Facades\DB;
+  use Spatie\Permission\Models\Role;
+
+  class CreateUsers extends Command
+  {
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'users:create {email} {password} {role} {--F|full_name=Default Name} {--A|address=Default Address} {--P|phone=0000000000}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Create a user have an email, password, and role...';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+      parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return void
+     * @throws Exception
+     */
+    public function handle()
+    {
+      $user['email'] = $this->argument('email');
+      $user['password'] = $this->argument('password');
+      $user['role'] = $this->argument('role');
+      $user['full_name'] = $this->option('full_name');
+      $user['address'] = $this->option('address');
+      $user['phone'] = $this->option('phone');
+
+      if(User::query()->where('email', '=', $user['email'])->exists())
+      {
+        $this->error('ERROR: The email: `'.$user['email'].'` has existed');
+        return;
+      }
+
+      if(!$this->isValidRole($user['role']))
+      {
+        $this->error('ERROR: The role `'.$user['role'].'` is invalid!');
+        return;
+      }
+
+      $this->insertUser($user);
+    }
+
+    private function isValidRole(string $role): bool
+    {
+      foreach (Role::all() as $r)
+      {
+        if ($role === $r->name)
+          return true;
+      }
```

Comment:
Tại sao ko check Role::where('name', $r->name)->exists(); ?? mà lại phải for all :(